### ISSUE TITLE
sometimes first job exit with 60303 file exist

### ISSUE THREAD

**Issue by ericvaandering** [dmwm/CRAB2#519]
Original Savannah ticket 65966 reported by farinafa on Fri Apr 16 09:59:27 2010.

see
https://hypernews.cern.ch/HyperNews/CMS/get/crabFeedback/3045/1/1/1/2/1/1/1/1/1/2/1.html
the issue is far from clear, but as I have now 3 users with same problem, it is most likely a real problem.
It may not be due to crab, but the solution may have to be found in crab. I will add more as I figure out.
Simply need a place to keep track.

**Comment by ericvaandering** [comment]
Comment by belforte on Thu Apr 29 01:24:38 2010

see latest entry in HN
https://hypernews.cern.ch/HyperNews/CMS/get/crabFeedback/3045/1/1/1/2/1/1/1/1/1/2/1/1/1/2/1/1/1/1/1/1.html

I am very very very confused as to how this is happening.

it ha nothing to do with server, i can reproduce with crab client 272pre4
UI config file is OK, i put print statements in BLite SchedulerGlite.py and verified that glite -submit command is called with a config file where retries are set to zero and that the jdl it passes to the command also has retries set to zero, yet a logginInfo after submission shoes a JDL with retries=3 and shallow-retry=10 and if I make a simple standalone test with submission i see that what i put in jdl is passed on to job in loggingInfo.

Bottom line, non ci sto' a capire una segaccia di nulla.
(english translation: it is pitch dark and I feel likely to be eaten by a grue).

Filippo, if you have a chance to also look it might help, as I am maybe too much confused to think clearly here, have a bunch of continous meetings today/tomorrow and am away on family business sunday/tuesday (at least).

Thanks

Stefano

**Comment by ericvaandering** [comment]
Comment by spigafi on Thu Apr 29 03:48:37 2010

I checked my mail archive and we exchanged mails on February 19 about that (me, you Stefano, DanieleS and Claudio). So, now the problem seems real (and quite pestering).

Firstly, I can check if something is changed on JDL definition (I already do that many times but... check again is not a bad idea). Then, we can try to verify if this problem is related to a specific WMS or not. If only one WMS is affected, the faster solution is to disable temporary it and contact the WMS support.

I suppose, the problem could be inside the WMS. If this is true, our actions to contain the problem are limited. I will bring you up to date if I discover something more.

**Comment by ericvaandering** [comment]
Comment by farinafa on Thu Apr 29 04:56:20 2010

CIao,
   Filippo and me found that there are two parameters for the JDL that crab does not override (probably they're newcomers with 3.2, Filippo is investigating). These take the default resubmission values from the UI glite_wms.conf, that infact sets resubmissions to 3 and 10 respectively.

I'm testing the fix for this issue.

Fabio &amp; Filippo

**Comment by ericvaandering** [comment]
Comment by spigafi on Thu Apr 29 05:09:20 2010

OK, I found an interesting thing going deeply into how WMS processes the JDL and what kind of information it puts inside the original JDL (the JDL submitted by CRAB) before it starts to process it. I will call &quot;WMS-JDL&quot; the one processed and filled by WMS and &quot;&quot;WMS-JDL-ORIG&quot;&quot; submitted by CRAB.

The facts are:
- &quot;glite-wms-job-info --jdl&quot; and &quot;glite-wms-job-info --jdl-original&quot; produce two different JDL, &quot;WMS-JDL-ORIG&quot; and &quot;WMS-JDL
- the JDL produced by &quot;glite-wms-job-info --jdl&quot; (WMS-JDL) contains for each job in the collection a replica of the requirements
- &quot;RetryCount&quot; and &quot;ShallowRetryCount&quot; for each job in the collection have the default value (3 and 10) ... this is wrong!
- &quot;RetryCount&quot; and &quot;ShallowRetryCount&quot; have the correct value but that value is related to the collection, not the job (this explain why resubmissions happen for single job/grid failure)

So, I think, two JDL parameters are missing inside the JDL (or missing inside the config under /afs/cern.ch/project/gd/LCG-share/3.2.4-0/glite/etc/cms/glite_wms.conf)
- DefaultNodeRetryCount = The DefaultNodeRetryCount attribute specifies value of the RetryCount attribute to be applied to all sub-jobs of the collection not specifying their own RetryCount.
- &quot;DefaultNodeShallowRetryCount&quot; = The DefaultNodeShallowRetryCount attribute specifies value of the ShallowRetryCount attribute to be applied to all sub-jobs of the collection not specifying their own ShallowRetryCount.
  (see JDL attribute guide for more details)

Now, what we can to is to try to force these two parameters before submit using CRAB and check if something changes inside JDL-WMS. If I reach a success, further actions must be arranged.

More news before the end of the day.

**Comment by ericvaandering** [comment]
Comment by farinafa on Thu Apr 29 05:17:31 2010

We solved it.

By adding the attributes described below by Filippo I'm able to have a JDL with Retry and Shallow-Retry actually set in accordance with the CRAB/user's will.

I just committed the fix.

Thanks to Filippo for the huge help in debugging!

**Comment by ericvaandering** [comment]
Comment by belforte on Thu Apr 29 05:28:00 2010

ah... cazzo..mi ero completamente dimenticato che ci fosse un JDL per la collection ed uno per i jobs etc. etc.
ma si', diamo tutta la colpa a Claudio !!
bravissimi, grazie !!

**Comment by ericvaandering** [comment]
Closed by farinafa on Tue May 11 07:07:43 2010
